sdk/lib/compartment_helpers/claim_fast.cc

```diff
@@ -0,0 +1,73 @@
+#include "cheri.hh"
+#include "compartment-macros.h"
+#include <cheriot-atomic.hh>
+#include <compartment.h>
+#include <errno.h>
+#include <futex.h>
+#include <stdlib.h>
+#include <switcher.h>
+
+int heap_claim_fast(Timeout *timeout, const void *ptr, const void *ptr2)
+{
+	void   **hazards = switcher_thread_hazard_slots();
+	auto    *epochCounter{const_cast<
+      cheriot::atomic<uint32_t> *>(MMIO_CAPABILITY_WITH_PERMISSIONS(
+	     cheriot::atomic<uint32_t>, allocator_epoch, true, false, false, false))};
+	uint32_t epoch  = epochCounter->load();
+	int      values = 2;
+	// Skip processing pointers that don't refer to heap memory.
+	if (!heap_address_is_valid(ptr))
+	{
+		ptr = nullptr;
+		values--;
+	}
+	if (!heap_address_is_valid(ptr2))
+	{
+		ptr2 = nullptr;
+		values--;
+	}
+	// If neither pointer refers to heap memory, set without synchronizing.  It
+	// doesn't matter if the revoker sees these or not, they cannot extend the
+	// lifetime of a heap object.
```

Comment:
Is this comment a bit stale? I don't understand it and I'm not sure it describes the code.

Reply:
No, this refers directly to the code immediately below it.  Above, we've checked whether either pointer refers to heap memory.  Below, if they don't, we null out the hazard slots and return immediately.